compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -908,8 +908,11 @@ void OMR::X86::CodeGenerator::deactivateDependentDiscardableRegisters(TR::Regist
 
 bool OMR::X86::CodeGenerator::supportsOpMaskRegisters()
 {
-    return self()->comp()->getOption(TR_EnableOpMaskRegisters)
-        && self()->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F);
+    TR::CPU *cpu = &self()->comp()->target().cpu;
+
+    return self()->comp()->getOption(TR_EnableOpMaskRegisters) && cpu->supportsFeature(OMR_FEATURE_X86_AVX512F)
+        && cpu->supportsFeature(OMR_FEATURE_X86_AVX512VL) && cpu->supportsFeature(OMR_FEATURE_X86_AVX512BW)
+        && cpu->supportsFeature(OMR_FEATURE_X86_AVX512DQ);
```

Comment:
Apologies, I should have asked this earlier.  Why are all these AVX512 features now required in order to support mask registers?